### ISSUE TITLE
Support for math operations on intermediate tensors

### ISSUE THREAD

**Issue by soumyac1999** [flexflow/FlexFlow#365]
**Context:**
I am implementing a VAE in FlexFlow. In VAEs, the encoder returns `(mean, cov)` of a Gaussian. A sample is drawn from this Gaussian and passed to the decoder. However, since sampling is not differentiable, the sampling is reparameterized as `mean + N(0, I) * cov`. For my particular implementation, the covariance is assume constant, hence the encoder outputs `mean` and `mean + random_normal(0, I) * constant` is passed to the decoder.

**Feature Request:**
Support for `constant * random_normal(0, I) + prev_layer_output` (possibly as a layer). Or suggestions on how to implement this using currently supported functions

**Comment by soumyac1999** [comment]
@lockshaw, I don't think I can add assignees or labels

**Comment by lockshaw** [comment]
@soumyac1999 I forget, where did we land on this issue?

**Comment by soumyac1999** [comment]
The particular operations which I have described above has been implemented. But will need to add a few other operations for the loss function. Should be close this and create a new issue?

Operations needed: 
| Operation | Status (Keras) | Status (FlexFlow Python) | Status (FlexFlow C++) |
| --- | --- | --- | --- |
| Slice | #363 | https://github.com/flexflow/FlexFlow/blob/66545f7239ec70504d0675615829c937a56783c7/python/flexflow/keras/utils/generic_utils.py#L528  |  |
| sin, cos | Need to Add | |
| Negation | Need to Add (?) | |
| Transpose | ? | https://github.com/flexflow/FlexFlow/blob/3652efb8916ccc8605402a8c38c4c513220fe903/python/flexflow/core/flexflow_cffi.py#L370 | https://github.com/flexflow/FlexFlow/blob/a43f8709baf7b644b0f6b5b6b0ebe1e1281698ed/include/flexflow/ops/transpose.h#L16 |
| BMM | ? | https://github.com/flexflow/FlexFlow/blob/3652efb8916ccc8605402a8c38c4c513220fe903/python/flexflow/core/flexflow_cffi.py#L342 | | 
| exp | ? | https://github.com/flexflow/FlexFlow/blob/3652efb8916ccc8605402a8c38c4c513220fe903/python/flexflow/core/flexflow_cffi.py#L101 | | 
| sqrt | ? | https://github.com/flexflow/FlexFlow/blob/3652efb8916ccc8605402a8c38c4c513220fe903/python/flexflow/core/flexflow_cffi.py#L279 | |

Keras (actual) does not expose many of these operations and these are made available through `keras.backend`. We might need to discuss if/how we would want to expose these in FlexFlow Keras.

**Comment by lockshaw** [comment]
@soumyac1999 Sure, that'd be great--can you create one for each operator that we need to work on and assign it to me?

You can close this issue once that's done.

**Comment by lockshaw** [comment]
@soumyac1999 Is this ready to be closed yet (i.e., have issues for the individual operators been created)?

**Comment by soumyac1999** [comment]
I would want to keep this issue open for some more time till I complete the loss function implementation if that's ok.
